fhir-search/src/main/java/com/ibm/fhir/search/uri/UriBuilder.java

```diff
@@ -75,20 +72,22 @@ public UriBuilder requestUri(String requestUriString) {
      * @throws URISyntaxException
      */
     public String toSearchSelfUri() throws URISyntaxException {
-
         URI requestUri = new URI(requestUriString);
 
         // Always include page size at the beginning, even if it wasn't in the request
         queryString.append(SearchConstants.COUNT);
         queryString.append(SearchConstants.EQUALS_CHAR);
         queryString.append(context.getPageSize());
-        queryString.append(SearchConstants.AND_CHAR);
 
         // the URLs is converted from a for loop to a functional paradigm. The functional
         // paradigm is consistent with JOINING multiple strings
-        Function<Parameter, String> serializeSearchParmToQueryString = p -> serializeSearchParmToQueryString(p);
-        queryString.append(context.getSearchParameters().stream().map(serializeSearchParmToQueryString).collect(Collectors.joining(SearchConstants.AND_CHAR_STR)));
-
+        if (!context.getSearchParameters().isEmpty()) {
```

Comment:
can context.getSearchParameters() be empty at this point? but anyway, I like this added check.

Reply:
It should never be null. (I tested )